lib/iris/plot.py

```diff
@@ -454,10 +453,6 @@ def _draw_2d_from_bounds(draw_method_name, cube, *args, **kwargs):
                 else:
                     values = coord.contiguous_bounds()
                     values = _fixup_dates(coord, values)
-                    if values.dtype == np.dtype(object) and isinstance(
-                        values[0], datetime.datetime
-                    ):
-                        values = mpl_dates.date2num(values)
```

Comment:
Been staring at these blocks for the past hour trying to work out WTF they are doing here. Nothing stops working without them - I ran all the tests locally - but as you say  we could be looking at some outdated defensive code. What is more confusing to me is that they completely undermine the upstream use of `_fixup_dates()` - why have both?!

I even tried introducing a non-standard calendar but still everything works without these blocks being here.

Reply:
Good point and 2199 is a nice find, too